lib/commands/aggregate.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+var _ = require('lodash');
+var util = require('util');
+
+var utils = require('../utils');
+var BaseCommand = require('./base');
+var ElementWrapper = require('../element-wrapper');
+
+var wrapElementForAccess = ElementWrapper.wrapElementForAccess;
+
+function Aggregate(server) {
+  BaseCommand.call(this, server);
+}
+BaseCommand.setUpCommand(Aggregate, 'aggregate');
+
+function getErrorReply(errorMessage, code) {
+  var error = {ok: false, errmsg: errorMessage};
+  if (code) {
+    error.code = code;
+  }
+  return {documents: [error]};
+}
+
+// We are implementing aggregation groups via objects. Objects in JavaScript
+// can only have strings as keys, but MongoDB aggregations can be based on all
+// kinds of JavaScript types.  In order to distinguish for example, between a
+// number 2 and a string '2', we need some extra logic which this function
+// provides.
+function buildValueKey(value) {
+  if (utils.isObjectId(value)) {
```

Comment:
Because the underlying checks are not scalar comparisons, there will always be a bunch of `if` statements hiding somewhere around.